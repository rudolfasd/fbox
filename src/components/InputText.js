import React from 'react';


class InputText extends React.Component {

	state = {text:''};

	hookEnter = e => {
		if (e.key !== 'Enter') return ;
 		this.setState({text:''});
		if (e.target.value){
			this.props.sagaAddPoint({text: e.target.value});
		};
	};

	handleInput = e => {
		this.setState({
			text: e.target.value
		});
	};
	
	render(){
		return(
			<div className='input-text'>
				<input className='input-text-field'
					type='text' maxLength='60'
					onKeyPress={this.hookEnter.bind(this)}
					onChange={this.handleInput}
					value={this.state.text}
				/>
			</div>
		);
	};
};

export default InputText;

