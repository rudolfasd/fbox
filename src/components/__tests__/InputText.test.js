import React from 'react';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import InputText from '../InputText';


describe('REACT COMPONENT <InputText /> desc1',() => {
    it('InputText - snapshot',() => {
        const renderedValue = renderer.create(
            <InputText />
        ).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });
});

describe('REACT COMPONENT <InputText /> desc2',() => {
    const mockSagaAddPoint = jest.fn();
    const mockProps = { sagaAddPoint: mockSagaAddPoint };
    let component;

    beforeEach(() => {
        mockSagaAddPoint.mockClear();
        component = shallow(<InputText {...mockProps}/>);
    });

    it('InputText - handleInput text entering',() => {
        const mockTextInputEvent = {target: {value: 'S'}, key:'S'};
        expect(component.instance().state).toEqual({text: ''});
        component.instance().handleInput(mockTextInputEvent);
        expect(component.instance().state).toEqual({text: 'S'});
    });
    it('InputText - hookEnter when enter pressed',() => {
        const mockEnterPressEvent = {
            target: {value: 'Some text ...'}, key: 'Enter'
        };
        component.instance().handleInput(mockEnterPressEvent);
        component.instance().hookEnter(mockEnterPressEvent);
        expect(component.instance().state).toEqual({text: ''});
        expect(mockSagaAddPoint).toHaveBeenCalledTimes(1);
        expect(mockSagaAddPoint).toHaveBeenCalledWith({ text: 'Some text ...' });
    });
    it('InputText - hookEnter when other key pressed',() => {
        const mockOtherKeyPressEvent = {
            target: {value: 'Some text ...OtherKey'}, key: 'OtherKey'
        };
        component.instance().handleInput(mockOtherKeyPressEvent);
        component.instance().hookEnter(mockOtherKeyPressEvent);
        expect(component.instance().state).toEqual({ text: 'Some text ...OtherKey' });
        expect(mockSagaAddPoint).toHaveBeenCalledTimes(0);
    });
    it('InputText - hookEnter when enter pressed but input field is empty',() => {
        const mockOtherKeyPressEvent = { target: {value: ''}, key: 'Enter'};
        component.instance().handleInput(mockOtherKeyPressEvent);
        component.instance().hookEnter(mockOtherKeyPressEvent);
        expect(component.instance().state).toEqual({ text: '' });
        expect(mockSagaAddPoint).toHaveBeenCalledTimes(0);
    });
});