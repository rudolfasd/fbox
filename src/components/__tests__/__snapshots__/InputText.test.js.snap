// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REACT COMPONENT <InputText /> desc1 InputText - snapshot 1`] = `
<div
  className="input-text"
>
  <input
    className="input-text-field"
    maxLength="60"
    onChange={[Function]}
    onKeyPress={[Function]}
    type="text"
    value=""
  />
</div>
`;
